modules/applications/pages/ui-builder/how-to-configure-https.adoc

```diff
@@ -0,0 +1,141 @@
+ssl_prefer_server_ciphers on;
+ssl_session_cache shared:SSL:50m;
+ssl_session_timeout 5m;
+add_header Strict-Transport-Security max-age=15768000;
+ssl_certificate /etc/ssl/certs/nginx.crt;  # ssl_certificate directive: Specifies the path to the SSL certificate.
+ssl_certificate_key /etc/ssl/private/nginx.key;  # ssl_certificate_key directive: Specifies the path to the SSL certificate key.
+ssl_protocols TLSv1.2 TLSv1.3;  # ssl_protocols directive: Specifies the supported SSL protocols.
+ssl_dhparam /etc/ssl/certs/dhparam.pem;  # ssl_dhparam directive: Specifies the path to the Diffie-Hellman parameter file.
+----
+
+=== Secure the cookies
+
+The secure flag on Bonita Runtime `X-Bonita-API-Token` and `JSESSIONID` and Bonita UI Builder `SESSION` cookies should be set when using HTTPS.
+Activating the secure flag can be done by updating the nginx configuration of Bonita UI proxy.
+Update `nginx.conf` file and modify the `proxy_cookie_path` directive for the locations `/bonita/sessionLogout`, `/bonita/` and `/bonita/authservice` as such:
+[source]
+----
+proxy_cookie_path /bonita "/; Secure";
+----
+Also add the following line to the locations `~* ^/app/([^?/#]+)` and `\`:
+[source]
+----
+proxy_cookie_path / "/; secure";
+----
+
+=== Redirect HTTP to HTTPS
+
+To redirect HTTP requests to HTTPS, you need to update the nginx configuration of Bonita UI proxy.
+Update `nginx.conf` file to add the following server block:
+
+[source]
+----
+server {
+    listen 80;
+    server_name localhost;
+    return 302 https://$server_name$request_uri;
+}
+----
+
+== Configure Bonita Runtime
+
+=== Configure the environment variables
+
+In order to replace the apparent scheme in requests to Bonita Runtime (HTTP) with the one of the browser (HTTPS), you need to set the environment variable `REMOTE_IP_VALVE_ENABLED: "true"` in the Bonita Runtime service in the `docker-compose.yml` file:
+[source,yaml]
+----
+services:
+  ...
+  bonita:
+    image: ...
+    environment:
+      REMOTE_IP_VALVE_ENABLED: "true"
+----
+
+== Troubleshooting
+
+*Symptom:* +
+302 Redirections are broken in Bonita applications and users get redircted to a URL without HTTPS
+
+*Cause:* +
+This issue arises because the incoming requests that are recieved by Bonita Runtime are HTTP requests and not HTTPS (since the reverse proxy is used for SSL offloading)
+
+*Solution:* +
+Activate the Remote IP Valve on Bonita container as described in the previous section in order for the `X-Forwarded-...` headers to be retrieved as request headers by Bonita Runtime.
```

Comment:
```suggestion
[.troubleshooting-title]
== Troubleshooting

[.troubleshooting-section]
--
[.symptom]
Navigation is broken in Bonita Applications

[.symptom-description]
302 Redirections are broken in Bonita applications and users get redirected to a URL without HTTPS

[.cause]#Cause#
This issue arises because the incoming requests that are recieved by Bonita Runtime are HTTP requests and not HTTPS (since the reverse proxy is used for SSL offloading)

[.solution]#Possible solution#
Activate the Remote IP Valve on Bonita container as described in the previous section in order for the `X-Forwarded-...` headers to be retrieved as request headers by Bonita Runtime.
--
```